Dockerfile

```diff
@@ -1,21 +1,74 @@
 # Docker demo image, as used on try.jupyter.org and tmpnb.org
 
-FROM jupyter/minimal-notebook:2d878db5cbff
+FROM jupyter/all-spark-notebook:a249876881d3
 
 MAINTAINER Jupyter Project <jupyter@googlegroups.com>
 
-# Install system libraries first as root
+# BEGININCLUDE jupyter/datascience-notebook
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+# FROM jupyter/scipy-notebook
+
+# MAINTAINER Jupyter Project <jupyter@googlegroups.com>
+
 USER root
 
-# Julia dependencies
-RUN apt-get update &&  \
-    apt-get install -y julia libnettle4 && \
-    apt-get clean
+# R pre-requisites
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    fonts-dejavu \
+    gfortran \
+    gcc && apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
-# R dependencies that conda can't provide (X, fonts, compilers)
+# Julia dependencies
 RUN apt-get update && \
-    apt-get install -y libxrender1 fonts-dejavu gfortran gcc && \
-    apt-get clean
+    apt-get install -y --no-install-recommends \
+    julia \
+    libnettle4 && apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+USER jovyan
+
+# R packages including IRKernel which gets installed globally.
+RUN conda config --add channels r && \
+    conda install --quiet --yes \
+    'rpy2=2.7*' \
+    'r-base=3.2*' \
+    'r-irkernel=0.5*' \
+    'r-plyr=1.8*' \
+    'r-devtools=1.9*' \
+    'r-dplyr=0.4*' \
+    'r-ggplot2=1.0*' \
+    'r-tidyr=0.3*' \
+    'r-shiny=0.12*' \
+    'r-rmarkdown=0.8*' \
+    'r-forecast=5.8*' \
+    'r-stringr=0.6*' \
+    'r-rsqlite=1.0*' \
+    'r-reshape2=1.4*' \
+    'r-nycflights13=0.1*' \
+    'r-caret=6.0*' \
+    'r-rcurl=1.95*' \
+    'r-randomforest=4.6*' && conda clean -tipsy
+
+# Install IJulia packages as jovyan and then move the kernelspec out
+# to the system share location. Avoids problems with runtime UID change not
+# taking effect properly on the .local folder in the jovyan home dir.
+RUN julia -e 'Pkg.add("IJulia")' && \
+    mv /home/$NB_USER/.local/share/jupyter/kernels/* $CONDA_DIR/share/jupyter/kernels/ && \
+    chmod -R go+rx $CONDA_DIR/share/jupyter && \
+    rm -rf /home/$NB_USER/.local/share
```

Comment:
Here's what's killing the toree kernel. Doesn't happen in stacks because the datascience image is separate.

This workaround was for dealing with a user permission ID when switching the UID of the jovyan user. Since that's probably not a requirement here in the demo image, might mean the julia setup should be done manually like the other kernels.